drodrpg/DRODLib/TotalMapStates.h

```diff
@@ -9,8 +9,8 @@
 //*******************************************************************************
 class CTotalMapStates {
 public:
-	CTotalMapStates();
-	~CTotalMapStates() { };
+	CTotalMapStates() { }
+	~CTotalMapStates() { }
 	void Load(const UINT dwPlayerID, const UINT dwHoldID);
 	MapState GetStoredMapStateForRoom(const UINT roomID);
```

Comment:
The trick is to make this method const, and then you can remove the const casts.

Reply:
This made no difference, it still refuses to let me call it from other classes. The only fix seems to be to stop making the CCurrentGame instance in MapWidget a const.

Reply:
> This doesn't work because TotalMapStates itself is not const inside CCurrentGame, and it can't be because we need to be able to update the room states. Even though this isn't relevant to the particular pathway of calling GetStoredMapStateForRoom, the compiler will not be happy unless all the object instances along the way are also const.

I'm probably not following the use cases here, but regarding the instance in MapWidget, if only const methods are called there, then that instance can remain const.

Reply:
Sorry, my bad, I did not understand that to make the method const you have to put the const keyword at the end of the definition, not the front.